qtgui/QtPixelator.cpp

```diff
@@ -1,244 +1,331 @@
-#include "QtPixelator.h"
-#include "error_codes.h"
-#include "logging.h"
-#include <vector>
-#include <optional>
-#include <cmath>
-#include <algorithm>
-#include <QPainter>
-
-namespace
-{
-    unsigned greatest_common_divisor(unsigned a, unsigned b);
-    QColor minDiff(const QColor& in_source, const std::vector<QColor>& in_list);
-}
- 
-QtPixelator::QtPixelator(QObject* in_parent)
-    : QObject(in_parent) 
-    , imageBuffer{}
-    , sourcePath{}
-    , storagePath{}
-    , stitchWidth{0}
-    , stitchHeight{0}
-    , stitchCount{0}
-    , rowCount{0}
-{}
-
-errors::Code QtPixelator::run(){
-    auto result = checkSettings();
-    if (errors::NONE == result)
-    {
-        QImage colorMap = pixelate();
-        if (colorMap.isNull()) return errors::PIXELATION_ERROR;
-        if (! scalePixels(colorMap)) return errors::PAINT_ERROR;
-        if (! imageBuffer.save(storagePath.toLocalFile())) result = errors::WRITE_ERROR;
-        logging::LogStream::instance().getLogStream(logging::Level::DEBUG) << "File written" << std::endl;
-    }
-    return result;
-}
-errors::Code QtPixelator::setInputImage(const QUrl& in_url)
-{
-    sourcePath = in_url;
-    logging::LogStream::instance().getLogStream(logging::Level::DEBUG) << "Load from " << in_url.toLocalFile().toStdString() << " on completion." << std::endl;
-    return (imageBuffer.load(in_url.toLocalFile()) ? errors::NONE : errors::WRONG_INPUT_FILE);
-}
-
-errors::Code QtPixelator::setOutputImage(const QUrl& in_url)
-{
-    logging::LogStream::instance().getLogStream(logging::Level::DEBUG) << "Store to " << in_url.toLocalFile().toStdString() << " on completion." << std::endl;
-    if ((sourcePath == in_url) && !(!sourcePath.isEmpty()))

```

Comment:
do? create issue? reinsert? explicitly abandon the idea because you're already asked on file selection if it exists?